### PR TITLE
LSan: Fix memory leak of test_libhttp2

### PR DESCRIPTION
**Pull request by masaori335**
Part of #10539.

### CODE REVIEW

File: proxy/http2/unit_tests/test_HpackIndexingTable.cc [hunk 1]
```diff
@@ -71,7 +71,10 @@ TEST_CASE("HPACK low level APIs", "[hpack]")
       HpackIndexingTable indexing_table(4096);
 
       for (const auto &i : indexed_test_case) {
-        std::unique_ptr<HTTPHdr> headers(new HTTPHdr);
+        std::unique_ptr<HTTPHdr, void (*)(HTTPHdr *)> headers(new HTTPHdr, [](HTTPHdr *hdr) {
+          hdr->destroy();
+          delete hdr;
```
Comment (masaori335) [inline proxy/http2/unit_tests/test_HpackIndexingTable.cc:76]: It looks like the destructor of `HTTPHdr` should call  `destroy()`, but it triggers a crash that the same with #10541.

**Comment by masaori335** [comment]
[approve ci autest]

File: proxy/http2/unit_tests/test_HpackIndexingTable.cc (hunk 1 above)
Comment (bneradt) [inline proxy/http2/unit_tests/test_HpackIndexingTable.cc:76]: This gets reused a few times in here. Can we:
1. Create a separate function for this.
2. Add a comment there explaining why this is destroy call is needed. (Basically `HTTPHdr` needs to be destroyed as well as deleted to be cleaned up properly).

**Review by bneradt — CHANGES_REQUESTED** [review]
Makes sense.

File: proxy/http2/unit_tests/test_HpackIndexingTable.cc (hunk 1 above)
Comment (masaori335) [reply proxy/http2/unit_tests/test_HpackIndexingTable.cc]: Done. Please take another look.

**Comment by masaori335** [comment]
@bneradt The issue is addressed. Please take another look.

**Review by bneradt — APPROVED** [review]

**Comment by zwoop** [comment]
Cherry-picked to v9.2.x

**Pull request merged**